src/serde/reader.rs

```diff
@@ -41,7 +41,11 @@ impl<R: Read, E: ByteOrder> Deserializer<R, E> {
     }
 
     fn read_bytes(&mut self, count: u64) -> Result<()> {
-        self.read += count;
+        if let (read, false) = self.read.overflowing_add(count) {
```

Comment:
My worry was a maliciously crafted length value causing an overflow panic. But you're right. People will have overflow checks disabled in their deployed release builds, which means an overflow won't cause their programs to terminate with a panic. I'll revert this. But be warned, once you add fuzzer tests to bincode, you'll find yourself adding this again. ;)